syntax = "proto3";

package common;

// 错误码 - 与内部 C++ ErrorCode 保持一致
enum ErrorCode {
    // ========== 成功 ==========
    OK = 0;

    // ==================== 通用错误（100~999） ====================
    // 系统错误（1xx）
    UNKNOWN             = 100;  // 未知错误
    INTERNAL            = 101;  // 内部服务器异常
    NOT_IMPLEMENTED     = 102;  // 功能未实现
    SERVICE_UNAVAILABLE = 103;  // 服务不可用
    TIMEOUT             = 104;  // 请求超时

    // 参数错误（2xx）
    INVALID_ARGUMENT    = 200;  // 参数无效
    INVALID_PAGE        = 210;  // 无效的分页参数
    INVALID_PAGE_SIZE   = 211;  // 无效的分页大小

    // 限流（3xx）
    RATE_LIMITED        = 300;  // 请求过于频繁
    QUOTA_EXCEEDED      = 301;  // 配额超限

    // ==================== 认证错误（1000~1999） ====================
    // Token 相关（100x）
    UNAUTHENTICATED     = 1000; // 未认证（通用）
    TOKEN_MISSING       = 1001; // Token 缺失
    TOKEN_INVALID       = 1002; // Token 无效
    TOKEN_EXPIRED       = 1003; // Token 已过期
    TOKEN_REVOKED       = 1004; // Token 已注销

    // 登录相关（101x~102x）
    LOGIN_FAILED        = 1010; // 登录失败（通用）
    WRONG_PASSWORD      = 1011; // 密码错误
    ACCOUNT_LOCKED      = 1012; // 账号已锁定（多次密码错误）

    // 验证码相关（102x）
    CAPTCHA_WRONG       = 1021; // 验证码错误
    CAPTCHA_EXPIRED     = 1022; // 验证码已过期

    // ==================== 用户错误（2000~2999） ====================
    // 用户查询（200x）
    USER_NOT_FOUND      = 2000; // 用户不存在
    USER_DELETED        = 2001; // 用户已删除

    // 用户创建（201x）
    USER_ALREADY_EXISTS = 2010; // 用户已存在（通用）
    MOBILE_TAKEN        = 2013; // 手机号已被占用

    // 用户状态（202x）
    USER_DISABLED       = 2020; // 用户已禁用
    USER_NOT_VERIFIED   = 2021; // 用户未验证

    // ==================== 权限错误（3000~3999） ====================
    PERMISSION_DENIED   = 3000; // 无权限（通用）
    ADMIN_REQUIRED      = 3001; // 需要管理员权限
    OWNER_REQUIRED      = 3002; // 需要是资源所有者
}


message Result {
    ErrorCode code              = 1;
    string    msg               = 2;
}